### ISSUE TITLE
Support TypeScript

### ISSUE THREAD

**Issue by gaslimitreached** [Developer-DAO/create-dao#27]
It is possible to support TypeScript:
- [nextjs](https://nextjs.org/docs/basic-features/typescript)
- [hardhat](https://hardhat.org/guides/typescript.html)

Steps:
- create `templates` directory
- rename `template` to `default`
- move `default` to `templates`
- create `templates/typescript` directory
  - will next/hardhat need separate directories due to`tsconfig.json`?
- update `create-app.js` for `--typescript`
  - install dependencies
  - copy `templates/typescript` assets

Reference for the steps above:
[create-next-app](https://github.com/vercel/next.js/tree/master/packages/create-next-app/templates)

**Comment by revmischa** [comment]
Why let people make non typescript projects at all? Nobody should be doing this in 2021 imo.

**Comment by e-roy** [comment]
correct, this is possible.  I left code for the option of typescript install [here](https://github.com/Developer-DAO/create-dao/blob/0709815bf658dee5ae747c3591acea46003cb1e1/index.js#L20)
https://github.com/Developer-DAO/create-dao/blob/0709815bf658dee5ae747c3591acea46003cb1e1/index.js#L20

and the code to copy the correct templated folder [here](https://github.com/Developer-DAO/create-dao/blob/0709815bf658dee5ae747c3591acea46003cb1e1/create-app.js#L138)
https://github.com/Developer-DAO/create-dao/blob/0709815bf658dee5ae747c3591acea46003cb1e1/create-app.js#L138

I think the idea was to focus a working plain js version as a MVP and exploring ts later.

**Comment by carlomigueldy** [comment]
I would volunteer to help migrate this project to TypeScript

**Comment by berteotti** [comment]
Should we add TypeScript for a template project?

**Comment by revmischa** [comment]
> I would volunteer to help migrate this project to TypeScript

Go for it! Give me typescript or give me death.

**Comment by e-roy** [comment]
Changes in the file structure coming down the pipeline.  This helps for smoother npm publishing and future scaling of this project if tested and merged.

[Linked PR](https://github.com/Developer-DAO/create-dao/pull/28)

**Comment by carlomigueldy** [comment]
> Changes in the file structure coming down the pipeline. This helps for smoother npm publishing and future scaling of this project if tested and merged.
> 
> [Linked PR](https://github.com/Developer-DAO/create-dao/pull/28)

Nicely done, please ping me here in this issue whenever it's ready to be migrated to TypeScript and hopeful I'm not gonna interfere the development cycle
